subsys/net/lib/sockets/sockets.c

```diff
@@ -1122,6 +1122,8 @@ Z_SYSCALL_HANDLER(zsock_getsockopt, sock, level, optname, optval, optlen)
 	kernel_optval = z_thread_malloc(kernel_optlen);
 	Z_OOPS(!kernel_optval);
 
+	memcpy(kernel_optval, (void *)optval, kernel_optlen);
```

Comment:
Consider using z_user_alloc_from_copy(), which will make this function much simpler.
definition is in kernel/include/syscall_handler.h

Reply:
Good point, uploaded new version that uses z_user_alloc_from_copy()